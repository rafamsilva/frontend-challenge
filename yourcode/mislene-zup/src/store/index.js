import Vue from 'vue'
import Vuex from 'vuex'
import createPersistedState from 'vuex-persistedstate'
import { createModule } from 'vuex-toast'
import 'vuex-toast/dist/vuex-toast.css'
import Movies from '@/store/modules/movies'
Vue.use(Vuex)

export default new Vuex.Store({
  // actions,
  modules: {
    Movies,
    // Account,
    // Queries,
    // Databases,
    // ExecuteQuery,
    toast: createModule({
      dismissInterval: 5000
    })
  },
  strict: false,
  plugins: [
    createPersistedState({
      key: 'vuex-zup'
    })
  ]
})
