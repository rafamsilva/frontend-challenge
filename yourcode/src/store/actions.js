export function MOVIES_FETCH_REQUESTED_ACTION(text) {
  return {
    type: 'MOVIES_FETCH_REQUESTED',
    payload: {
      text,
    },
  };
}
