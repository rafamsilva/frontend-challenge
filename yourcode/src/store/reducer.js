const INITIAL_STATE = {
  lastFilter: '',
  loading: false,
  movies: [],
};

export default function reducer(state = INITIAL_STATE, action) {
  switch (action.type) {
    case 'START_LOADER':
      return Object.assign({}, state, {
        loading: true,
      });
    case 'STOP_LOADER':
      return Object.assign({}, state, {
        loading: false,
      });
    case 'SET_FILTER':
      return Object.assign({}, state, {
        lastFilter: action.payload.filter,
      });
    case 'MOVIES_FETCH_SUCCEEDED':
      return Object.assign({}, state, {
        loading: false,
        movies: action.payload.movies,
      });
    case 'MOVIES_FETCH_FAILED':
      return Object.assign({}, state, {
        loading: false,
        err: action.payload.err,
      });
    default:
      return state;
  }
}
