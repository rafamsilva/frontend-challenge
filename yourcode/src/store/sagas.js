import { call, put, takeLatest } from 'redux-saga/effects';
import OmdbProvider from '../providers/omdb.provider';

function* fetchMovies(action) {
  const filter = action.payload.text;

  if (filter) {
    try {
      yield put({ type: 'START_LOADER' });
      yield put({ type: 'SET_FILTER', payload: { filter } });

      const movies = yield call(OmdbProvider.fetchMovies, filter);

      yield put({ type: 'MOVIES_FETCH_SUCCEEDED', payload: { movies } });
    } catch (err) {
      yield put({ type: 'MOVIES_FETCH_FAILED', payload: { err } });
    }
  }
}

export function* fetchMoviesSaga() {
  yield takeLatest('MOVIES_FETCH_REQUESTED', fetchMovies);
}
