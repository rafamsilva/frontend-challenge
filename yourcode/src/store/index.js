import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';

import reducer from './reducer';
import { fetchMoviesSaga } from './sagas';

const sagaMiddleware = createSagaMiddleware();

export default createStore(reducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(fetchMoviesSaga);
