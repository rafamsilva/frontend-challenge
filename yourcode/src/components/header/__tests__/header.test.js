import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import { store } from '../../../store';

import Header from '../';

describe('Header Component', () => {
  let container;

  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  describe('Render', () => {
    it('should render without crashing', () => {
      act(() => {
        render(
          <Router>
            <Header showSearchBar={false} />
          </Router>,
          container,
        );
      });
    });

    it('should render without search bar', () => {
      act(() => {
        render(
          <Router>
            <Header showSearchBar={false} />
          </Router>,
          container,
        );
      });

      expect(container.querySelector("[data-testid='search-bar']")).toBeNull();
    });

    it('should render with search bar', () => {
      act(() => {
        render(
          <Provider store={store}>
            <Router>
              <Header showSearchBar={true} />
            </Router>
          </Provider>,
          container,
        );
      });

      expect(container.querySelector("[data-testid='search-bar']")).toBeDefined();
    });
  });

  describe('Snapshot', () => {
    it('should match snapshot without search bar', () => {
      act(() => {
        render(
          <Router>
            <Header showSearchBar={false} />
          </Router>,
          container,
        );
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it('should match snapshot with search bar', () => {
      act(() => {
        render(
          <Provider store={store}>
            <Router>
              <Header showSearchBar={true} />
            </Router>
          </Provider>,
          container,
        );
      });

      expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
  });
});
