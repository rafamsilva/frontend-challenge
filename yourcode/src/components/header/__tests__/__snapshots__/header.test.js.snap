// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot should match snapshot with search bar 1`] = `
"<header class=\\"sc-AxhCb kBBbbl\\"><a class=\\"sc-AxhUy fpQjAw\\" href=\\"/\\"><svg alt=\\"What's in Logo\\" title=\\"What's in Logo\\">logo.svg</svg></a>
  <div class=\\"sc-AxjAm dUDgzT\\"><svg alt=\\"Icon\\" title=\\"Icon\\" class=\\"sc-AxirZ hAQDsX\\">icon-magnifier-grey.svg</svg><input placeholder=\\"Search movies...\\" class=\\"sc-AxiKw exrimN\\" type=\\"text\\" value=\\"\\"></div>
</header>"
`;

exports[`Header Component Snapshot should match snapshot without search bar 1`] = `"<header class=\\"sc-AxhCb kBBbbl\\"><a class=\\"sc-AxhUy fpQjAw\\" href=\\"/\\"><svg alt=\\"What's in Logo\\" title=\\"What's in Logo\\">logo.svg</svg></a></header>"`;
